enhancements/samples-generation.md

```diff
@@ -0,0 +1,292 @@
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+1. Should not the project has a Memcached GO Sample with webhooks and other without using webhooks since it shows an advanced feature and it used is not mandatory? Note the Go tutorial do not use webhooks.  
+
+## Summary
+
+This proposal is over the samples getting started to be generated on the SDK repo via go to bring more flexibilities and facilities to keep them maintained and improve their implementation and customizations. 
+
+By using Golang, we can easily inject code implementation as have helpers which brings a lot of facilities to customize them. Also, it allows a more readable implementation.  
+
+## Motivation
+
+- Improve the maintainability, readability and reusability
+- None effort to keep the samples updated with the latest releases
+- Less effort and more flexibility to customize the samples  
+
+### Goals
+
+- Centralize the samples in the [operator-sdk-samples](https://github.com/operator-framework/operator-sdk-samples) repository generated on SDK repo
+- For each PR a validation should be made to ensure that all samples are update
+- Provide the SampleContext as part of the generation of the sample on SDK repo at the same way that provides the [TestContext](https://github.com/kubernetes-sigs/kubebuilder/blob/master/test/e2e/utils/test_context.go#L36) 
+- Allow to the e2e tests and samples uses the same helpers to mock the data 
+- Archive the repository [operator-sdk-samples](https://github.com/operator-framework/operator-sdk-samples)
+- Facilitate the review process and the development process since is possible verify the result of the code changes by running the make target to update the samples as check them in the PR's
+
+### Non-Goals
+
+- Implement tests to ensure that the samples are working as expected
+- Link the samples in the docs or aligned then with the tutorials
+
+## Proposal
+
+### I am as maintainer/contributor would like to:
+
+- be able to troubleshoot the code used to create the samples and easy improve them as solve issues that might be faced
+- no longer spend effort to update the samples on the release process
+- easily able to create samples based on the tutorial steps
+- ensure that the changes performed against the master will not introduce bugs in the steps required to build the samples and following up the tutorials
+- able to implement the code to generate the samples using the same approach used in the e2e tests (it is very helpful to keep the project maintained)
+- able to easily review pr's which has changes for the scaffolds (without the need to clone them locally)
+      
+### I am as operator developer would like to:
+- check the examples of projects and features provided to help me build my operators as know how they work
+- check the project which is the result of the tests performed in the tutorials provided
+
+### Risks and Mitigations
+
+N/A
```

Comment:
There are no risks to generating complex example operators? What about copying parts of the reconcile loop into the generated operator and the copied code isn't updated?

Reply:
The build of the samples is tested by the CI. However, is a NO GO run tests over the samples to ensure that they are runnable. Regards the inject code is wrong then, it is the risk of any code/change implementation against the project but OK, we can add that.